cfme/tests/openstack/infrastructure/test_roles.py

```diff
@@ -1,39 +1,85 @@
-pytestmark = [pytest.mark.uncollectif(lambda: current_version() < '5.7'),
-              pytest.mark.usefixtures("setup_provider_modscope")]
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
 
 
 ROLES = ['NovaCompute', 'Controller', 'Compute', 'BlockStorage', 'SwiftStorage',
          'CephStorage']
 
 
-def test_host_role_type(provider):
-    navigate_to(provider, 'Details')
-    sel.click(InfoBlock.element("Relationships", "Nodes"))
-    my_quads = list(Quadicon.all())
-    assert len(my_quads) > 0
-    for quad in my_quads:
-        role_name = str(quad.name)
-        role_name = re.search(r'\((\w+)\)', role_name).group(1)
-        assert role_name in ROLES
+def test_host_role_association(provider, soft_assert):
+    navigate_to(Host, 'All')
+    names = [q.name for q in list(Quadicon.all())]
+    for node in names:
+        host = Host(node, provider)
+        navigate_to(host, 'Details')
+        role_name = summary_title().split()[1].translate(None, '()')
+        role_name = 'Compute' if role_name == 'NovaCompute' else role_name
+        role_assoc = host.get_detail('Relationships', 'Deployment Role')
+        soft_assert(role_name in role_assoc, 'Deployment roles misconfigured')
 
 
-def test_roles_name(provider):
+def test_roles_name():
     navigate_to(DeploymentRoles, 'All')
     my_roles_quads = list(Quadicon.all())
     for quad in my_roles_quads:
         role_name = str(quad.name).split('-')[1]
         assert role_name in ROLES
+
+
+def test_roles_summary(provider, soft_assert):
+    err_ptrn = '{} are shown incorrectly'
+    navigate_to(DeploymentRoles('', provider), 'AllForProvider')
+    roles_names = [q.name for q in Quadicon.all()]
+
+    for name in roles_names:
+        dr = DeploymentRoles(name, provider)
+        navigate_to(dr, 'DetailsFromProvider')
+
+        values = ('Nodes', 'Direct VMs', 'All VMs')
+        for v in values:
+            res = dr.get_detail('Relationships', v)
+            soft_assert(res.isdigit(), err_ptrn.format(v))
+
+        values = ('Total CPUs', 'Total Node CPU Cores')
+        for v in values:
+            res = dr.get_detail('Totals for Nodes', v)
+            soft_assert(res.isdigit() and int(res) > 0, err_ptrn.format(v))
+
+        total_cpu = dr.get_detail('Totals for Nodes', 'Total CPU Resources')
+        soft_assert('GHz' in total_cpu, err_ptrn.format('Total CPU Resources'))
+        total_memory = dr.get_detail('Totals for Nodes', 'Total Memory')
+        soft_assert('GB' in total_memory, err_ptrn.format('Total Memory'))
+
+        values = ('Total Configured Memory', 'Total Configured CPUs')
+        for v in values:
+            res = dr.get_detail('Totals for VMs', v)
+            soft_assert(res, err_ptrn.format(v))
+
+
+def test_role_delete(provider):
+    navigate_to(DeploymentRoles, 'All')
+    quads = list(Quadicon.all())
+    assert quads > 0
+    role_name = choice(quads).name
+    dr = DeploymentRoles(role_name, provider)
+    dr.delete()
```

Comment:
Some node (or nodes), which deployment role will have been deleted, loses a relation to a deployment role, so provider will not know whether particular node is a controller or compute or storage.. I believe it won't affect other tests since the deletion of role is the last test for roles at all. But if there is a way to make it run at the end of the whole test execution - I'd like to use it